velox/functions/sparksql/MakeTimestamp.cpp

```diff
@@ -187,18 +188,18 @@ std::shared_ptr<exec::VectorFunction> createMakeTimestampFunction(
     const std::vector<exec::VectorFunctionArg>& inputArgs,
     const core::QueryConfig& config) {
   const auto sessionTzName = config.sessionTimezone();
-  VELOX_USER_CHECK(
+  VELOX_CHECK(
       !sessionTzName.empty(),
       "make_timestamp requires session time zone to be set.")
   const auto sessionTzID = util::getTimeZoneID(sessionTzName);
 
   const auto& secondsType = inputArgs[5].type;
-  VELOX_USER_CHECK(
+  VELOX_CHECK(
```

Comment:
@mbasmanova With user check used, the fuzzer won't fail if invalid parameters are supplied. I feel VELOX_CHECK should be used in this case to reject invalid parameters. I can extract these changes to a separate PR if you agree. Thanks.

Reply:
@rui-mo Rui, this is an interesting question. In Velox, we distinguish two classes of errors: (1) User error is an error caused by the user of the library; to make the error go away, the user needs to make some changes in how they use the library; (2) System error indicates a bug in Velox or some failure in the environment. 

The errors here seems to fall into "User error" class. 

Would you explain why this is causing a problem in the Fuzzer? Is it because Fuzzer assumes that any error can be suppressed by TRY? Do we need to make it possible to tell whether a given User error can be suppressed by TRY or not?

Reply:
Thanks for explaining the difference between user and system error. In the fuzzer test, user errors are caught and treated as acceptable if canThrow is true ([link](https://github.com/facebookincubator/velox/blob/main/velox/expression/tests/ExpressionVerifier.cpp#L136-L143)).
Here I'm thinking whether custom arg generators are needed for make_timestamp and unscaled_value functions. They both throw user error for long decimal type. Without arg generators, `ArgumentTypeFuzzer::fuzzArgumentTypes` generates random decimal types, whose precision can exceed 18. If using arg generators, it is ensured that only short decimal type can be generated. However, as you mentioned, user errors are more proper here, it seems we can continue to generate random decimal types and test both the error and correct paths. If so, I can follow up in a separate PR to remove custom arg generator for the two function. Thanks!

Reply:
@rui-mo The Fuzzer has --retry-with-try option that re-runs failing expression by wrapping it in TRY and expects the expression to succeed. I assume this won't work if we generate invalid input types for make_timestamp and unscaled_value functions. Hence, it seems like we need to continue to use custom arg generators.

Reply:
Thanks for the pointer. Indeed, the fuzzer test does not work without the custom arg generators when enabling `--retry_with_try` option. In this PR, added custom arg generators for them and enabled `--retry_with_try` option in the workflow test.